test/powershell/Modules/Microsoft.PowerShell.Management/FileSystem.Tests.ps1

```diff
@@ -246,6 +246,140 @@ Describe "Basic FileSystem Provider Tests" -Tags "CI" {
+        New-Item -ItemType SymbolicLink $secondSymToOther -Value $otherSubDir >$null
+        New-Item -ItemType SymbolicLink $symToSym -Value $symToOther >$null
+        New-Item -ItemType SymbolicLink $symToOtherFile -Value $otherFile >$null
+        New-Item -ItemType HardLink $hardToOtherFile -Value $otherFile >$null
+
+        if ($IsWindows)
+        {
+            New-Item -ItemType Junction $junctionToOther -Value $otherSubDir >$null
+            New-Item -ItemType SymbolicLink $symdToOther -Value $otherSubDir >$null
+        }
+    }
+
+    Context "Copy-Item using different case (on case-sensitive file systems)" {
+        BeforeEach {
+            $sourcePath = $filePath
+            $destinationPath = "$TestDrive/" + $fileName.Toupper()
+        }
+        AfterEach {
+            Remove-Item -Path $destinationPath -ErrorAction SilentlyContinue
+        }
+
+        It "Copy-Item can copy to file name differing only by case" {
+            if ($isCaseSensitive)
+            {
+                Copy-Item -Path $sourcePath -Destination $destinationPath -ErrorAction SilentlyContinue | Should Be $null
+                Test-Path -Path $destinationPath | Should Be $true
+            }
+            else
+            {
+                { Copy-Item -Path $sourcePath -Destination $destinationPath -ErrorAction Stop } | ShouldBeErrorId "CopyError,Microsoft.PowerShell.Commands.CopyItemCommand"
+                $Error[0].Exception | Should BeOfType System.IO.IOException
+                $Error[0].Exception.Data[$selfCopyKey] | Should Not Be $null
+            }
+        }
+    }
+
+    Context "Copy-Item avoids copying an item onto itself" {
+        BeforeAll {
+            $testCases = @(
+                @{
+                    Name = "Copy to same path"
+                    Source = $otherFile
+                    Destination = $otherFile
+                }
+                @{
+                    Name = "Copy hard link"
+                    Source = $hardToOtherFile
+                    Destination = $otherFile
+                }
+                @{
+                    Name = "Copy hard link, reversed"
+                    Source = $otherFile
+                    Destination = $hardToOtherFile
+                }
+                @{
+                    Name = "Copy symbolic link to target"
+                    Source = $symToOtherFile
+                    Destination = $otherFile
+                }
+                @{
+                    Name = "Copy symbolic link to symbolic link with same target"
+                    Source = $secondSymToOther
+                    Destination = $symToOther
+                }
+                @{
+                    Name = "Copy through chain of symbolic links"
+                    Source = $symToSym
+                    Destination = $otherSubDir
+                }
+            )
+
+            # Junctions and directory symbolic links are Windows and NTFS only
+            if ($IsWindows)
```

Comment:
Oh, we shouldn't skip tests silently 😕 
Maybe define the test body (below in "It" block) as function and make two "It" blocks (second "skip on non-Windows")?